### ISSUE TITLE
[yugabyted] Crash when using a dash in YSQL_USER environment variable

### ISSUE THREAD

**Issue by timothyvanderaerden** [yugabyte/yugabyte-db#6410]
With YugaByteDB version 2.5.0, `yugabyted` crashes when using a dash `-` in the username of the `YSQL_USER` environment variable. For example:
```
docker run -e YSQL_USER=some-user yugabytedb/yugabyte:latest  yugabyted start --daemon=false
```

Log:

```[yugabyted start] 2020-11-18 17:57:39,605 INFO:  | 0.0s | About to start master with cmd /home/yugabyte/bin/yb-master --stop_on_parent_termination --undefok=stop_on_parent_termination --fs_data_dirs=/home/yugabyte/var/data --webserver_interface=0.0.0.0 --metrics_snapshotter_tserver_metrics_whitelist=handler_latency_yb_tserver_TabletServerService_Read_count,handler_latency_yb_tserver_TabletServerService_Write_count,handler_latency_yb_tserver_TabletServerService_Read_sum,handler_latency_yb_tserver_TabletServerService_Write_sum,disk_usage,cpu_usage,node_up --yb_num_shards_per_tserver=1 --ysql_num_shards_per_tserver=1 --cluster_uuid=ea813145-72a4-44ae-884f-eac7ecf844df --rpc_bind_addresses=127.0.0.1:7100 --server_broadcast_addresses=127.0.0.1:7100 --replication_factor=1 --use_initial_sys_catalog_snapshot --server_dump_info_path=/home/yugabyte/var/data/master-info --master_enable_metrics_snapshotter=true --webserver_port=7000 --default_memory_limit_to_ram_ratio=0.35 --instance_uuid_override=72b5592fe7804df99585ad6eeee1deba --master_addresses=127.0.0.1:7100
[yugabyted start] 2020-11-18 17:57:39,607 INFO:  | 0.0s | master started running with PID 17.
[yugabyted start] 2020-11-18 17:57:39,608 INFO:  | 0.0s | Waiting for master
[yugabyted start] 2020-11-18 17:57:39,608 INFO:  | 0.0s | run_process: cmd: [u'/home/yugabyte/bin/yb-admin', u'--init_master_addrs', u'127.0.0.1:7100', u'list_all_masters']
[yugabyted start] 2020-11-18 17:57:39,719 INFO:  | 0.1s | run_process returned 1: 
OUT >>

<< ERR >>
<<
[yugabyted start] 2020-11-18 17:57:40,220 INFO:  | 0.6s | run_process: cmd: [u'/home/yugabyte/bin/yb-admin', u'--init_master_addrs', u'127.0.0.1:7100', u'list_all_masters']
[yugabyted start] 2020-11-18 17:57:59,273 INFO:  | 19.7s | run_process returned 0: 
OUT >>
Master UUID                      	RPC Host/Port        	State    	Role
72b5592fe7804df99585ad6eeee1deba 	127.0.0.1:7100       	ALIVE    	LEADER

<< ERR >>
<<
[yugabyted start] 2020-11-18 17:57:59,273 INFO:  | 19.7s | Querying for all masters in cluster
[yugabyted start] 2020-11-18 17:57:59,273 INFO:  | 19.7s | Waiting to get the full master addrs list from master
[yugabyted start] 2020-11-18 17:57:59,273 INFO:  | 19.7s | run_process: cmd: [u'/home/yugabyte/bin/yb-admin', u'--init_master_addrs', u'127.0.0.1:7100', u'list_all_masters']
[yugabyted start] 2020-11-18 17:57:59,571 INFO:  | 20.0s | run_process returned 0: 
OUT >>
Master UUID                      	RPC Host/Port        	State    	Role
72b5592fe7804df99585ad6eeee1deba 	127.0.0.1:7100       	ALIVE    	LEADER

<< ERR >>
<<
[yugabyted start] 2020-11-18 17:57:59,571 INFO:  | 20.0s | Got all masters: [u'127.0.0.1:7100']
[yugabyted start] 2020-11-18 17:57:59,572 INFO:  | 20.0s | Got full master addrs list: [u'127.0.0.1:7100']
[yugabyted start] 2020-11-18 17:57:59,572 INFO:  | 20.0s | Old master flag is: [u'--tserver_master_addrs=127.0.0.1:7100'] and new master flag is: --tserver_master_addrs=127.0.0.1:7100
[yugabyted start] 2020-11-18 17:57:59,572 INFO:  | 20.0s | About to start tserver with cmd /home/yugabyte/bin/yb-tserver --stop_on_parent_termination --undefok=stop_on_parent_termination --fs_data_dirs=/home/yugabyte/var/data --webserver_interface=0.0.0.0 --metrics_snapshotter_tserver_metrics_whitelist=handler_latency_yb_tserver_TabletServerService_Read_count,handler_latency_yb_tserver_TabletServerService_Write_count,handler_latency_yb_tserver_TabletServerService_Read_sum,handler_latency_yb_tserver_TabletServerService_Write_sum,disk_usage,cpu_usage,node_up --yb_num_shards_per_tserver=1 --ysql_num_shards_per_tserver=1 --cluster_uuid=ea813145-72a4-44ae-884f-eac7ecf844df --rpc_bind_addresses=0.0.0.0:9100 --server_broadcast_addresses=127.0.0.1:9100 --cql_proxy_bind_address=0.0.0.0:9042 --server_dump_info_path=/home/yugabyte/var/data/tserver-info --start_pgsql_proxy --pgsql_proxy_bind_address=0.0.0.0:5433 --tserver_enable_metrics_snapshotter=true --metrics_snapshotter_interval_ms=11000 --webserver_port=9000 --default_memory_limit_to_ram_ratio=0.6 --instance_uuid_override=0076d524f1d24ac4b48cc13f21bd14dd --start_redis_proxy=false --tserver_master_addrs=127.0.0.1:7100
[yugabyted start] 2020-11-18 17:57:59,585 INFO:  | 20.0s | tserver started running with PID 91.
[yugabyted start] 2020-11-18 17:57:59,587 INFO:  | 20.0s | Waiting for tserver ...
[yugabyted start] 2020-11-18 17:57:59,588 INFO:  | 20.0s | run_process: cmd: [u'/home/yugabyte/bin/yb-admin', u'--init_master_addrs', u'127.0.0.1:7100', u'list_all_tablet_servers']
[yugabyted start] 2020-11-18 17:57:59,854 INFO:  | 20.3s | run_process returned 0: 
OUT >>
Tablet Server UUID               RPC Host/Port Heartbeat delay Status   Reads/s  Writes/s Uptime   SST total size  SST uncomp size SST #files      Memory   

<< ERR >>
<<
[yugabyted start] 2020-11-18 17:58:00,354 INFO:  | 20.8s | run_process: cmd: [u'/home/yugabyte/bin/yb-admin', u'--init_master_addrs', u'127.0.0.1:7100', u'list_all_tablet_servers']
[yugabyted start] 2020-11-18 17:58:00,484 INFO:  | 20.9s | run_process returned 0: 
OUT >>
Tablet Server UUID               RPC Host/Port Heartbeat delay Status   Reads/s  Writes/s Uptime   SST total size  SST uncomp size SST #files      Memory   
0076d524f1d24ac4b48cc13f21bd14dd 78d699d53677:9100 0.39s           ALIVE    0.00     0.00     0        0 B             0 B             0               7.34 MB  

<< ERR >>
<<
[yugabyted start] 2020-11-18 17:58:00,485 INFO:  | 20.9s | Found tserver uuid 0076d524f1d24ac4b48cc13f21bd14dd in list of tserver uuids [u'0076d524f1d24ac4b48cc13f21bd14dd'].
[yugabyted start] 2020-11-18 17:58:00,485 INFO:  | 20.9s | Completed waiting for tserver.
[yugabyted start] 2020-11-18 17:58:00,696 INFO:  | 21.1s | run_process: cmd: [u'/home/yugabyte/bin/ysqlsh', u'-h', '127.0.0.1', u'-p', '5433', u'-c', u'\\conninfo']
[yugabyted start] 2020-11-18 17:58:00,701 INFO:  | 21.1s | run_process returned 2: 
OUT >>

<< ERR >>
<<
[yugabyted start] 2020-11-18 17:58:00,701 INFO:  | 21.1s | Unable to connect using YSQL. Error: ysqlsh: could not connect to server: Connection refused
	Is the server running on host "127.0.0.1" and accepting
	TCP/IP connections on port 5433?

[yugabyted start] 2020-11-18 17:58:01,201 INFO:  | 21.6s | run_process: cmd: [u'/home/yugabyte/bin/ysqlsh', u'-h', '127.0.0.1', u'-p', '5433', u'-c', u'\\conninfo']
[yugabyted start] 2020-11-18 17:58:01,678 INFO:  | 22.1s | run_process returned 0: 
OUT >>
You are connected to database "yugabyte" as user "yugabyte" on host "127.0.0.1" at port "5433".

<< ERR >>
<<
[yugabyted start] 2020-11-18 17:58:01,678 INFO:  | 22.1s | YSQL Connection Info - You are connected to database "yugabyte" as user "yugabyte" on host "127.0.0.1" at port "5433".

[yugabyted start] 2020-11-18 17:58:01,678 INFO:  | 22.1s | Setting up custom credentials for YSQL...
[yugabyted start] 2020-11-18 17:58:01,679 INFO:  | 22.1s | run_process: cmd: [u'/home/yugabyte/bin/ysqlsh', u'-h', '127.0.0.1', u'-p', '5433', u'-q', u'-c', u'\\t', u'-c', u"select datname from pg_catalog.pg_database where datname='some-user'"]
```

**Comment by tedyu** [comment]
Not sure if the following command was right (starting local cluster):
```
yugabyte-2.5.0.0 YSQL_USER=some-user ./bin/yugabyted start
Starting yugabyted...
✅ System checks

+--------------------------------------------------------------------------------------------------+
|                                            yugabyted                                             |
+--------------------------------------------------------------------------------------------------+
| Status              : Running                                                                    |
| Web console         : http://127.0.0.1:7000                                                      |
| JDBC                : jdbc:postgresql://127.0.0.1:5433/some-user?user=some-user&password=some-user |
| YSQL                : bin/ysqlsh   -U some-user -d some-user                                     |
| YCQL                : bin/ycqlsh   -u cassandra                                                  |
```
I was able to use ysqlsh to query the DB afterwards

**Comment by bhavin192** [comment]
https://github.com/yugabyte/yugabyte-db/pull/6420 fixed this issue and the change is available in v2.5.1. Please re-open if the issue persist.
